server/acceptor.hpp

```diff
@@ -0,0 +1,63 @@
+#ifndef ACCEPTOR_HPP
+#define ACCEPTOR_HPP
+
+#include <boost/asio.hpp>
+#include <atomic>
+
+namespace fizyr
+{
+
+namespace basio = boost::asio;
+
+/**
+ * \brief The Acceptor accepts a port number on which it will listen
+ *        for the incoming connection requests.
+ */
+class acceptor
+{
+public:
+    /**
+     * \brief acceptor constructor
+     * \param ios instance of 'io_service' class is required by socket constructor.
+     * \param port_num port number on which it will listen for the incoming connection
+     */
+    acceptor(basio::io_service &ios, unsigned short port_num);
```

Comment:
iiuc rule of 5 correctly, shouldn't you also have the remaining 4 defined along with the constructor?